crypto/include/internal/evp_int.h

```diff
@@ -21,6 +21,9 @@ struct evp_pkey_ctx_st {
     EVP_KEYEXCH *exchange;
     void *exchprovctx;
 
+    EVP_SIGNATURE *signature;
+    void *sigprovctx;
+
```

Comment:
This turns out interesting...  can an `EVP_PKEY_CTX` be used for multiple operations (i.e. is the `operation` field a bit field rather than a number)?  If not, we might want to consider a union here...

Reply:
No - I think you can only do one operation at a time.

Reply:
That's what I though...

Reply:
I changed this to a union - although it turned out to be quite complicated.

Reply:
I can imagine.  Suddenly, you become *much* more dependent on the `operation` field.  Feel free to declare that not worth the while if you prefer...

Reply:
... oh, and having to create a small struct for each union variant, of course...!

Reply:
Please take a look at feda77d and see what you think. I expect that there will be at least one more member in this union moving forward (possibly 2).